### PR TITLE
Add beauty salons

### PR DESCRIPTION
**Pull request by Vgoylo**

### CODE REVIEW

File: app/controllers/api/v1/salons_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Api::V1::SalonsController < ApplicationController
+  def index
+    salons = Salon.all
+    render json: salons, status: 200
```
Comment (blackp0is0n) [inline app/controllers/api/v1/salons_controller.rb:4]: @Vgoylo Returning status here is redundant

File: app/controllers/api/v1/salons_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def show
+    salon = Salon.find_by(id: params[:id])
+    if salon
+      render json: salon, status: 200
+    else
+      render json: { error: 'Error creating review.' }
```
Comment (blackp0is0n) [inline app/controllers/api/v1/salons_controller.rb:12]: @Vgoylo Status? There should not be 200

File: app/controllers/api/v1/salons_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Api::V1::SalonsController < ApplicationController
+  def index
+    salons = Salon.all
+    render json: salons, status: 200
+  end
+
+  def show
+    salon = Salon.find_by(id: params[:id])
+    if salon
+      render json: salon, status: 200
```
Comment (blackp0is0n) [inline app/controllers/api/v1/salons_controller.rb:10]: @Vgoylo There status is redundant

File: app/models/salon.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class Salon < ApplicationRecord
+  validates :name, length: { minimum: 2 }, uniqueness: true
+end
```
Comment (blackp0is0n) [inline app/models/salon.rb:3]: NL

File: config/routes.rb [hunk 5]
```diff
@@ -1,8 +1,9 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  namespace :api do
+    namespace :v1 do
+      resources :salons, only: [:index, :show]
```
Comment (blackp0is0n) [inline config/routes.rb:6]: @Vgoylo What about `update` and `destroy`?

**Comment by Vgoylo** [comment]
@blackp0is0n Can I add `update` && `destroy` in separate PR?

**Review by GEK62 — APPROVED** [review]

File: app/controllers/api/v1/salons_controller.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
+class Api::V1::SalonsController < ApplicationController
+  def index
+    @salons = Salon.all
+    render json: salons
+  end
+
+  def show
+    @salon = Salon.find_by(id: params[:id])
```
Comment (Rub1st) [inline app/controllers/api/v1/salons_controller.rb:8]: Please use find instead of find_by

File: app/controllers/api/v1/salons_controller.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    if salon
+      render json: @salon
+    else
+      render json: { error: 'Error creating review.' }
+    end
+  end
+
+  def create
+    @salon = Salon.new(
```
Comment (Rub1st) [inline app/controllers/api/v1/salons_controller.rb:17]: Probably you can write it as Salon.new(salon_params), I'm not sure, but it should work. check it please

**Review by Rub1st — COMMENTED** [review]
Looking good. Please fix issues that I left and I will approve this PR

**Review by Rub1st — APPROVED** [review]

**Review by GEK62 — APPROVED** [review]

**Pull request merged**